src/Http/Http/test/Features/RequestBodyPipeFeatureTests.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.IO.Pipelines;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Http.Features
+{
+    public class RequestBodyPipeFeatureTests
+    {
+        [Fact]
+        public void RequestBodyReturnsStreamPipeReader()
+        {
+            var context = new DefaultHttpContext();
+            var expectedStream = new MemoryStream();
+            context.Request.Body = expectedStream;
+
+            var provider = new RequestBodyPipeFeature(context);
+
+            var pipeBody = provider.RequestBodyPipe;
+
+            Assert.True(pipeBody is StreamPipeReader);
```

Comment:
This is not the greatest Assert. We care less about what type pipeBody and more that it functions.

Initialize `expectedStream` with data and read it from pipe.

Reply:
I'll add a separate set of test for those scenarios. I think both scenarios are important.